core/.js/src/main/scala/io/udash/utils/FileService.scala

```diff
@@ -0,0 +1,172 @@
+    URL.createObjectURL(blob)
+  }
+
+  /**
+   *  Converts specified bytes arrays to string that contains URL
+   *  that representing the array given in the parameter with `application/octet-stream` mime-type.
+   *
+   *  Keep in mind that returned URL should be revoked via `org.scalajs.dom.revokeObjectURL(url)`.
+   */
+  def asURL(bytesArrays: Seq[Array[Byte]]): String =
+    asURL(bytesArrays, OctetStreamType)
+
+  /**
+   *  Converts specified bytes array to string that contains URL
+   *  that representing the array given in the parameter with specified mime-type.
+   *
+   *  Keep in mind that returned URL should be revoked via `org.scalajs.dom.revokeObjectURL(url)`.
+   */
+  def asURL(byteArray: Array[Byte], mimeType: String): String =
+    asURL(Seq(byteArray), mimeType)
+
+  /**
+   *  Converts specified bytes array to string that contains URL
+   *  that representing the array given in the parameter with `application/octet-stream` mime-type.
+   *
+   *  Keep in mind that returned URL should be revoked via `org.scalajs.dom.revokeObjectURL(url)`.
+   */
+  def asURL(byteArray: Array[Byte]): String =
+    asURL(Seq(byteArray), OctetStreamType)
+
+  /**
+   * Create an anchor element that on click downloads specified URL as a file with specified name.
+   */
+  def asAnchor(filename: String, url: String): JsDom.TypedTag[Anchor] = {
+    import JsDom.all._
+
+    val download = attr("download")
+    a(href := url, download := filename)
+  }
+
+  /**
+   * Create an anchor element that on click downloads byte array as a file with specified name.
+   *
+   * Keep in mind that anchor's href URL should be revoked via `org.scalajs.dom.revokeObjectURL(url)`.
+   */
+  def asAnchor(filename: String, bytes: Array[Byte], mimeType: String = OctetStreamType): JsDom.TypedTag[Anchor] =
+    asAnchor(filename, asURL(Seq(bytes), mimeType))
+
+  /**
+   * Asynchronously convert specified file to bytes array.
+   */
+  def asBytesArray(file: File): Future[Array[Byte]] = {
+    import js.typedarray._
+
+    val fileReader = new FileReader()
+    val promise = Promise[Array[Byte]]()
+
+    fileReader.onerror = (e: Event) =>
+      promise.failure(new IOException(e.toString))
+
+    fileReader.onabort = (e: Event) =>
+      promise.failure(new IOException(e.toString))
+
+    fileReader.onload = (_: UIEvent) =>
+      promise.complete(Try(
+        new Int8Array(fileReader.result.asInstanceOf[ArrayBuffer]).toArray
+      ))
+
+    fileReader.readAsArrayBuffer(file)
+
+    promise.future
+  }
+
+  /**
+   * Convert specified file to InputStream with blocking I/O
+   *
+   * Because it is using synchronous I/O that could potentially this API can be used only inside worker.
+   *
+   * This method is using FileReaderSync that is part of Working Draft File API.
+   * Anyway it is supported for majority of modern browsers
+   */
+  def asInputStream(file: File, bufferSize: Int = 1024): InputStream =
```

Comment:
If your intention is just to process a file in chunks and you're already resorting to the working drafts, maybe using `Blob.stream()` would be a better option? The availability is not that poor (https://caniuse.com/?search=Blob.stream) and you could still benefit from async processing. I'm not convinced I want to ship synchronous API with input streams here - we'd have to provide more sane alternatives, which can be handled on the library client side more accurately.

Reply:
@ddworak it doesn't work on Safari :(

Reply:
I do have two reason to not use `Blob.stream`:
 1. it doesn't work at safari and safari mobile and this two browsers are in the supported scope of project that requires it :)
 2. a scala library that is using this api implements `InputStream` and I need to make a converter between `Blob.stream` and `InputStream` anyway.

=> I decided to move by simpler and most universal way.

And I don't think that forcing someone to use worker to huge job like parsing file is bad idea.

I'd love to say that it is good.

Reply:
I understand your motives, but I'm not convinced we should ship such a complex and use-case specific utility here. Let's strip this method and the whole input stream impl from this PR.

Reply:
@ddworak do you have anything in mind regarding `asBytesArraySync(...): Array[Byte]`?

Reply:
No, I think that will be cool!

Reply:
@ddworak I just tried to read 100 mb file via this way.

It is a bad idea. Very bad idea :)

So, why you think that `InputStream` is bad way?

Reply:
`asSyncBytesArray(file, start, end)` looks quite safe for large file

Reply:
Don't get me wrong, buffering is obviously essential for synchronous reading, I'm just not convinced Udash users will benefit from this since:
- 90% cases can be covered directly by `FileReader` and
- the other 10% cases can be covered by adding chunking to it (similar to e.g. https://stackoverflow.com/a/28318964)
- I don't want to encourage blocking the main thread

I hope Java stdlib port evolves enough to support such use cases out of the box (for SJS) https://github.com/scala-js/scala-js/tree/master/javalib/src/main/scala/java/io.